cdap-data-fabric/src/main/java/co/cask/cdap/store/OwnerStore.java

```diff
@@ -0,0 +1,207 @@
+          metaTable.put(createRowKey(entityId), COL, Bytes.toBytes(principal));
+        }
+      });
+    } catch (TransactionFailureException e) {
+      throw Transactions.propagate(e, IOException.class);
+    }
+  }
+
+  /**
+   * Retrieves the owner information for the given {@link EntityId}
+   *
+   * @param entityId the {@link EntityId} whose owner principal information needs to be retrieved
+   * @return {@link String} which is the principal of the {@link EntityId} owner
+   * @throws IOException if failed to get the store
+   */
+  public String get(final EntityId entityId) throws IOException {
+    try {
+      return Transactions.execute(transactional, new TxCallable<String>() {
+        @Override
+        public String call(DatasetContext context) throws Exception {
+          byte[] principalBytes = getTable(context).get(createRowKey(entityId), COL);
+          return principalBytes == null ? null : Bytes.toString(principalBytes);
+        }
+      });
+    } catch (TransactionFailureException e) {
+      throw Transactions.propagate(e, IOException.class);
+    }
+  }
+
+  /**
+   * Checks if owner information exists or not
+   *
+   * @param entityId the {@link EntityId} for which the check needs to be performed
+   * @return a boolean true: owner principal exists, false: no owner principal exists
+   * @throws IOException if failed to get the store
+   */
+  public boolean exists(final EntityId entityId) throws IOException {
+    try {
+      return Transactions.execute(transactional, new TxCallable<Boolean>() {
+        @Override
+        public Boolean call(DatasetContext context) throws Exception {
+          byte[] principalBytes = getTable(context).get(createRowKey(entityId), COL);
+          return principalBytes != null;
+        }
+      });
+    } catch (TransactionFailureException e) {
+      throw Transactions.propagate(e, IOException.class);
+    }
+  }
+
+  /**
+   * Deletes the owner principal for the given {@link EntityId}
+   *
+   * @param entityId the entity whose owner principal needs to be deleted
+   * @throws IOException if failed to get the owner store
+   */
+  public void delete(final EntityId entityId) throws IOException {
+    try {
+      transactional.execute(new TxRunnable() {
+        @Override
+        public void run(DatasetContext context) throws Exception {
+          getTable(context).delete(createRowKey(entityId));
+        }
+      });
+    } catch (TransactionFailureException e) {
+      throw Transactions.propagate(e, IOException.class);
+    }
+  }
+
+  private Table getTable(DatasetContext context) throws IOException, DatasetManagementException {
+    return DatasetsUtil.getOrCreateDataset(context, datasetFramework, DATASET_ID, Table.class.getName(),
+                                           DATASET_PROPERTIES);
+  }
+
+  // creates rowkey for association entries
+  private byte[] createRowKey(EntityId entityId) {
+    return Bytes.toBytes(OWNER_PREFIX + ':' + entityId.toString());
```

Comment:
Working on this currently. This needs some refactoring of existing code. We do something similar to build metadata key. If we pull it out we can make it common for storing entityid as key.